src/Evolution/Systems/Cce/BoundaryData.hpp

```diff
@@ -77,6 +75,102 @@ void cartesian_to_spherical_coordinates_and_jacobians(
     const Scalar<DataVector>& sin_phi, const Scalar<DataVector>& sin_theta,
     double extraction_radius) noexcept;
 
+/*
+ * \brief Compute \f$g_{i j}\f$, \f$g^{i j}\f$, \f$\partial_i g_{j k}\f$, and
+ * \f$\partial_t g_{i j}\f$ from input (potentially modal Spherepack) spatial
+ * metric quantities.
+ *
+ * \details This function interpolates the modes of
+ * input \f$g_{ij}\f$, \f$\partial_r g_{i j}\f$, and \f$\partial_r g_{i j}\f$ to
+ * the libsharp-compatible grid. This function then applies the necessary
+ * jacobian factors and angular derivatives to determine the full \f$\partial_i
+ * g_{j k}\f$. If `radial_renormalize` is set, this function will apply a
```

Comment:
Let's not support a SpEC bug, just fix SpEC output

Reply:
But without this, there's parts of the catalog that can't be extracted with SpECTRE...

Reply:
That's insane. The catalog should have the data corrected, not SpECTRE correcting it.

Reply:
I don't disagree that the catalog should be changed here, but in a practical sense the additional computational cost of this function is vastly lower than re-running the full merger simulation on all cases prior to the SpEC bugfix, which could be done, but probably won't be anytime soon.
The main usefulness of this will be as the catalog is run with the modern SpECTRE CCE, which includes additional outputs that SpEC CCE cannot provide. If you insist on not having this option in SpECTRE develop, the only practical alternative I can see is my personally maintaining a branch on my fork and routing people to that instead when they need to run on legacy SpEC output. That would be fine, just a bit of additional work to keep it appropriately rebased periodically.

Reply:
A few things I don't understand:

1. Since you can a posteriori change things in the SpECTRE CCE why can you not a posteriori edit the data in the catalog to account for this normalization?
2. How many simulations are we talking?

Reply:
1. I suppose it is possible; It would mean more work, but I could probably develop an independent python script to make the change. The irritation is that some of the spherical harmonics dealings that are easy in SpECTRE now because the spherepack library is entirely wrapped could be difficult when building something else from scratch. Not impossible, though.
2. Good question. Looks like the SpEC issue was closed Dec. 13 2017 (Issue 1097). So, presumably all simulations run prior to that date, which I'm not sure on.

Reply:
1. Hmm, would it be easy to add writing functionality to SpECTRE to write the CCE catalog format? I assume you've written code to read that in already. I certainly believe that doing the transformations in python code would be a pain....
2. Okay, maybe @markscheel knows? I have no intuition for which BFI projects have been run since that fix.

Reply:
It would be pretty easy to add the writing capability to SpECTRE. I'm coding that in anyways as a test helper to test the h5 reading functions. I'd be happy to just move the renormalization to its own tiny worldtube data fixer executable, if that's your suggestion?

Reply:
Yea, I would be happy with an executable in SpECTRE that updates the CCE catalag :) I'm sure we will need that type of utility in the future for SpECTRE bugs that show up anyway :P I think we should basically try to process file formats from other codes in our "primary" executables, then have separate utility codes for dealing with things like "Well this data is just wrong". Of course, if we can just change the catalog then we might not need the utility after it runs once

Reply:
@markscheel 's had the suggestion for this to incorporate all of the steps that deal with modes more directly into the h5 reading utility, and include the radial renormalization in those steps under the philosophy that the h5 reader is already unavoidably tailored to SpEC worldtube format.

I think that either creating a separate converter executable or moving it into the reader would be comparable amounts of work, so I'm happy to do either -- I just wanted to give voice to the alternative strategies.

Reply:
I think it's better to update the catalog to be consistent and correct since otherwise someone might in the future use the data outside of SpECTRE and be very confused with their results. :man_shrugging: